File: app/src/main/java/org/metabrainz/mobile/data/sources/QueryUtils.java
package org.metabrainz.mobile.data.sources;

import android.util.Pair;

import java.util.List;

public class QueryUtils {

    public static String getQuery(List<Pair<String, String>> arguments) {
        StringBuilder queryBuilder = new StringBuilder();
        for (Pair<String, String> argument : arguments)
            if (argument.second != null && !argument.second.isEmpty())
                queryBuilder.append("(").append(argument.first).append(":")
                        .append(argument.second).append(")");

        return queryBuilder.toString();

    }
}

File: app/src/main/java/org/metabrainz/mobile/data/sources/QueryUtils.kt
package org.metabrainz.mobile.data.sources

import kotlin.collections.HashMap

object QueryUtils {

    fun getQuery(arguments: HashMap<String, String>): String {
        val queryBuilder = StringBuilder()
        for ((key, value) in arguments){
            if (value.isNotEmpty()){
                queryBuilder.append("(").append(key).append(":").append(value).append(")")
            }
        }
        return queryBuilder.toString()
    }
}